Tmedia 113: atttribution hyphen fix

farleyawapo opened the pull request:
**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
This Pull Request modifies the blockquote and pullquote blocks. Citations for these types of quote blocks were being rendered with empty content(author), so the code if modified to:

1. not render a hyphen when there is no citation content
2. modify the css to keep styles clean and even across blocks
3. include tests

## Jira Ticket
- [TMEDIA-113](https://arcpublishing.atlassian.net/browse/TMEDIA-113)

## Acceptance Criteria
Actual Behavior
If users include a block or pull quote within Composer and they don't use the attribution field, on the front end a hyphen still appears within the quote as if there were an attribution.

Expected Behavior
If there is no attribution associated with the quote, there should be no hyphen. Ideally, this piece of the quote element would collapse.

## Test Steps
1. Checkout this branch `git checkout {branch name}`
2. Run fusion repo with linked blocks `npx fusion start -f -l {blocks to link}`
3. Run tests and check visually
4. Go to https://corecomponents.arcpublishing.com/pf/local/2021/04/26/pull-quote-block-quote-test-story/?_website=the-gazette to see original without corrections
5. Go to http://localhost/pf/local/2021/04/26/pull-quote-block-quote-test-story/?_website=the-gazette to see local branch changes

## Effect Of Changes
### Before
Hyphen rendered with no author attached
Pullquote:
![Screen Shot 2021-04-27 at 4 01 46 PM](https://user-images.githubusercontent.com/82830431/116312342-edeff080-a771-11eb-9dc8-58c2ec93a979.png)
Blockquote:
![Screen Shot 2021-04-27 at 4 02 27 PM](https://user-images.githubusercontent.com/82830431/116312393-ff38fd00-a771-11eb-8038-dc9a16e6cd5b.png)


### After
Hyphen not rendered with empty citation content(no author)
Pullquote:
![Screen Shot 2021-04-27 at 3 46 40 PM](https://user-images.githubusercontent.com/82830431/116312059-93569480-a771-11eb-850a-73786225e7c1.png)
Blockquote:
![Screen Shot 2021-04-27 at 3 46 57 PM](https://user-images.githubusercontent.com/82830431/116312131-a9fceb80-a771-11eb-8f6c-96ca470e247f.png)


## Dependencies or Side Effects
N/A

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

farleyawapo committed:
Removes hyphen when a quote attribution does not have content.

farleyawapo committed:
updated to type checking

farleyawapo committed:
corrections for quote styles after hyphen removal

farleyawapo committed:
test for modified block and pull quote citations

farleyawapo committed:
tests updated for blockquote

farleyawapo committed:
takes out space in scss

farleyawapo committed:
Merge branch 'canary' into TMEDIA-113_atttribution_hyphen_fix

matthewroach commented on blocks/article-body-block/chains/article-body/_articlebody.scss:
```diff
@@ -251,6 +254,11 @@
       }
     }
 
+    > * + * {
```
@farleyawapo - This is just a placement of CSS within the file, you can fix this by moving where the CSS is placed in the file

farleyawapo committed:
css lint corrections

farleyawapo committed:
removes tab

farleyawapo committed:
Merge branch 'canary' into TMEDIA-113_atttribution_hyphen_fix

JackHowa commented on blocks/article-body-block/chains/article-body/_articlebody.scss:
```diff
@@ -261,6 +265,11 @@
       text-align: left;
     }
+
+    > * + * {
```
@farleyawapo this seems like a pretty easy fix, no? ordering. some of this functionality comes automatically with vs code plugins for stylelint

farleyawapo replied:
That's what I thought I did according to the rule there, which is move it to where it should be, I put the most specific selector at the bottom of the code block. The linting for SCSS is still confusing to me and I don't think is consistent between the local development checks and Github hooks.

JackHowa commented on blocks/article-body-block/chains/article-body/_children/pullquote.jsx:
```diff
@@ -11,7 +11,7 @@ export default ({ element }) => {
     }
   });
-  if (citation.type === 'text') {
+  if (citation.type === 'text' && citation.content !== (null || undefined || '')) {
```
Not sure I totally follow this conditional. if the `citation.content` is `null`, we should not add the citation, right?

JackHowa replied:
```suggestion
  if (citation.type === 'text' && citation.content !== null  && citation.content !== undefined && citation.content !== '') {
```

JackHowa commented on blocks/article-body-block/chains/article-body/_children/pullquote.jsx:
```diff
@@ -11,7 +11,7 @@ export default ({ element }) => {
     }
   });
-  if (citation.type === 'text') {
+  if (citation.type === 'text' && citation.content !== (null || undefined || '')) {
```
<img width="322" alt="Screen Shot 2021-04-28 at 13 44 35" src="https://user-images.githubusercontent.com/5950956/116456379-d5421200-a827-11eb-9d59-9ed0309f0b96.png">

that `null || undefined || ''` will evaluate as "" because null and undefined are both falsy

<img width="292" alt="Screen Shot 2021-04-28 at 13 45 10" src="https://user-images.githubusercontent.com/5950956/116456449-ebe86900-a827-11eb-9414-7add54918520.png">

JackHowa replied:
I always look at this chart to doublecheck falsy values https://dorey.github.io/JavaScript-Equality-Table/

JackHowa reviewed (dismissed):
Good job adding test cases. Looks like there's the goal of testing for whether content is undefined or null. but that conditional will only test for empty string as far as I can see.

farleyawapo committed:
Update blocks/article-body-block/chains/article-body/_children/pullquote.jsx

Yes, I was trying to cover all the types since we don't have any type checking here, and I wasn't sure what types get passed to this block.

Co-authored-by: Jack Howard <jack.howard@washpost.com>

JackHowa commented:
@farleyawapo oh damn, my suggested code change had a linting error ... https://github.com/WPMedia/fusion-news-theme-blocks/pull/856/checks?check_run_id=2460589816#step:7:14 

I guess that's why it's better to code locally

farleyawapo committed:
removes extra space

Commit:
style: Follow stylelint descending specificity rec

farleyawapo committed:
fix styles linting and conditional expressions

farleyawapo committed:
Merge branch 'TMEDIA-113_atttribution_hyphen_fix' of github.com:WPMedia/fusion-news-theme-blocks into TMEDIA-113_atttribution_hyphen_fix

farleyawapo committed:
styling update

matthewroach reviewed (dismissed):
All works as intended. Nice work @farleyawapo 

We should create a ticket to clean up the CSS here for `blockquote` as there is a lot of duplication happening between the two different types.

farleyawapo committed:
Fixes linting problems while keeping style solutions

farleyawapo committed:
padding bottom resized for pullquote

matthewroach commented on blocks/article-body-block/chains/article-body/_articlebody.scss:
```diff
@@ -284,6 +220,79 @@
+      }
+    }
+
+    *:not(:first-child) {
```
You don't need this CSS here as the CSS you have on line 196 does the same

farleyawapo committed:
removes duplicate code

matthewroach reviewed (approved):
Nice work @farleyawapo ... 👍 🌮
